dashboard/app/helpers/levels_helper.rb

```diff
@@ -170,30 +171,32 @@ def app_options
       view_options(authored_hint_view_requests_url: authored_hint_view_requests_path(format: :json))
     end
 
-    if @level.is_a? Blockly
-      @app_options = blockly_options
-    elsif @level.is_a?(DSLDefined) || @level.is_a?(FreeResponse)
-      @app_options = question_options
-    elsif @level.is_a? Widget
-      @app_options = widget_options
-    elsif @level.unplugged?
-      @app_options = unplugged_options
+    if level.is_a? Blockly
+      @app_options[level.id] = blockly_options level
+    elsif level.is_a?(DSLDefined) || level.is_a?(FreeResponse)
+      @app_options[level.id] = question_options level
+    elsif level.is_a? Widget
+      @app_options[level.id] = widget_options level
+    elsif level.unplugged?
+      @app_options[level.id] = unplugged_options
     else
       # currently, all levels are Blockly or DSLDefined except for Unplugged
-      @app_options = view_options.camelize_keys
+      @app_options[level.id] = view_options.camelize_keys
     end
+    reset_view_options
```

Comment:
A few comments related to the freeze / reset:
* This seems sketchy to me because you are resetting and regenerating the view options, but whichever level you load last is going to get cached as `@view_options` and referenced by things like `@view_options.readonly_workspace` in `_single_text_match.html.haml`. If your levels differ on that property, you would get a wrong result.
* https://www.pivotaltracker.com/n/projects/914728/stories/98153794 says that we simply shouldn't be freezing view options in the first place
* perhaps if we are setting appOptions.level as an array `appOptions.levels` this problem will become moot